### DISCUSSION TITLE
How do I find the reasoning behind changes?

### DISCUSSION THREAD

**Discussion by fds-github** [rectorphp/rector#8725] [General]
I just started using Rector to check/fix my code for PHP 8, but I'm finding that it's doing a lot of general improvement along the way.  What a great tool!

Is there a way to find the explanation/justification behind rules?  Most of they time it's self-evident, but sometimes it's not clear to me.  The particular example I'm looking at now is StaticCallOnNonStaticToInstanceCallRector.  My assumption would have been that instantiating the class to make a call is less efficient and doesn't add any value.  I also assume I'm wrong, since Rector incorporated this rule, but how can I learn the error of my reasoning?  I've looked at rector_rules_overview.md, but the explanation is minimal.

Thanks for any help, and moreover all the work on a great tool!

**Comment by samsonasik** [comment]
`StaticCallOnNonStaticToInstanceCallRector` is fixing Non-static method cannot be called statically which deprecated on php <=7.x, and fatal error on php 8+ see https://3v4l.org/MFv93

**Reply by fds-github** [reply]
Thanks for the lightning-fast response.

What's the easiest way for me to find these answers for myself in the future?  I can imagine I could have learned this with an hour or so of googling, but that gets expensive as I pour through all the rules that I'm triggering.  I was hoping that there would be a succinct description like yours for each of the rules that gets added.

**Reply by samsonasik** [reply]
there was `@changelog` usage, that removed at :

- https://github.com/rectorphp/rector-src/pull/6035

some removed `@changelog` to php rfc seems needs to be moved to `@see` into `Rector\ValueObject\PhpVersionFeature` class constants :)

**Reply by fds-github** [reply]
Oh well.  Thanks anyway.
